nats.go

```diff
@@ -4190,12 +4190,26 @@ func (m *Msg) RespondMsg(msg *Msg) error {
 	if m.Reply == "" {
 		return ErrMsgNoReply
 	}
-	msg.Subject = m.Reply
```

Comment:
fyi @scottf @ColinSullivan1 this could happen as well in Java/.NET APIs with RespondMsg